pkg/operator/controllers/monitoring/monitoring_controller.go

```diff
@@ -109,8 +93,8 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		configData.PrometheusK8s.Retention = ""
 		changed = true
 	}
-	if configData.PrometheusK8s.VolumeClaimTemplate.Spec.Resources.Requests.Storage != "" {
-		configData.PrometheusK8s.VolumeClaimTemplate.Spec.Resources.Requests.Storage = ""
+	if !reflect.DeepEqual(configData.PrometheusK8s.VolumeClaimTemplate, struct{ api.MissingFields }{}) {
+		configData.PrometheusK8s.VolumeClaimTemplate = struct{ api.MissingFields }{}
```

Comment:
The line above it checks for Retention (default to 15 days). If we are going to make this configMap `prometheusK8s: {}`, I think we can remove this because any change to retention will trigger to make the empty object in the configmap.
```go
if configData.PrometheusK8s.Retention != "" {		
    configData.PrometheusK8s.Retention = ""		
    changed = true
}
```

Reply:
While I question managing retention the focus of this PR is to remove the volume claim template.  I think we should have a follow on discussion on if we manage this CM at all moving forward and if we do what parts we do manage.

Reply:
Agreed.  We had a discussion and we can resolve this comment.